### PR TITLE
A whole lot of markdown improvements

### PR DESCRIPTION
**Pull request by pacak**
@ematipico - can you check if this works for you or do you see any problems in generated markdown?

### CODE REVIEW

**Comment by ematipico** [comment]
Here are some suggestions that could help:
1. The usage could be wrapped inside a **code block**

	```shell
	app command
	```
	I would render nicely!
1. I would put the positional items at the top. They are a command's **mandatory** arguments, so it's best to show them first to the reader. The options are all optional, so they are slightly less important.
1. the `--help` output uses the documentation attached to a type as a description. While this is **very** useful, there might be a case where the doc comment goes on multiple lines or even contain code blocks. In the Rome repository, we usually read the first paragraph of a doc comment and remove the rest. That paragraph is then used as a brief description. You could use the same trick here. This suggestion takes me to the next one.
3. Support for markdown code blocks, so people can write some examples if they need to, to highlight some... I don't know... a particular case, usage, etc. That would be nice for the next release!

File: Changelog.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Change Log
 
+## bpaf [0.9.2] - Unreleased
+- with `docgen` feature you can render documentation as markdownc
```
Comment (ematipico) [inline Changelog.md:4]:
```suggestion
- with `docgen` feature you can render documentation as markdown
```

**Review by ematipico — APPROVED** [review]

**Comment by pacak** [comment]
> The usage could be wrapped inside a code block

For simple examples - yes, for longer examples this will require me to deal with breaking text into lines and what's more importantly - it will strip all the styling I added. Currently things are bold, italic, etc.

> I would put the positional items at the top.

Yea, that might work better. Interestingly enough none of the apps I looked at display info about positional items at all, `rustc`, `cargo`, `gcc`, `ls`...

> the --help output uses the documentation attached to a type as a description. 

I define paragraphs as something separated by an empty line, first paragraph shows in `--help` message,  everything shows up in `--help --help` (or `-hh`) or in the generated documentation.

> Support for markdown code blocks, so people can write some examples if they need to, to highlight some...

Codeblock is something aligned with 4 spaces and I guess I support it now...

**Comment by pacak** [comment]
>    I would put the positional items at the top.

Actually positional items are at the top but anything with `group_help` block takes the priority... :thinking:

**Comment by ematipico** [comment]
> For simple examples - yes, for longer examples this will require me to deal with breaking text into lines and what's more importantly - it will strip all the styling I added. Currently things are bold, italic, etc.
> 

Oh, I haven't considered that! I suppose we can leave it like this for now.

> I define paragraphs as something separated by an empty line, first paragraph shows in `--help` message,  everything shows up in `--help --help` (or `-hh`) or in the generated documentation.
> 

I tried it locally, and all paragraphs were included in the `--help` output. The code blocks too. Maybe I missed something. 

> > Support for markdown code blocks, so people can write some examples if they need to, to highlight some...
> 
> Codeblock is something aligned with 4 spaces and I guess I support it now...

Oh, ok. I'll try again. I was wondering if it could be rendered with the triple

```md
```sh
```

So it's rendered as actual shell command line

**Comment by pacak** [comment]
> I tried it locally, and all paragraphs were included in the --help output. The code blocks too. Maybe I missed something.

Can you push it somewhere?

> I was wondering if it could be rendered with the triple

Yea, looking into that - `rustdoc` is convinced that anything in a code block without annotation is rust code and fails to compile :)

**Comment by ematipico** [comment]
Here you can see the emitted help output: https://github.com/rome/tools/pull/4671/files#diff-09420bb6877d56d47ead6ac9021e9e227bb2ab0bae7d8f5e1c78aa03446c189b

Here's the change: https://github.com/rome/tools/pull/4671/files#diff-cbbce4bacb2713e24caf79ed1cb9757d2d23d7b6e96000b52946cd6c12033222

I noticed that the code block is rendered, but it doesn't have the language type anymore. 

Here's the rendered result: https://deploy-preview-4671--docs-rometools.netlify.app/cli/#rome-format

**Comment by pacak** [comment]
I see, you are trying to use ``` in the comments. Let's see what it takes to support that too :) The goal would be to strip it from the console output but keep in markdown...

**Comment by pacak** [comment]
Seems to work now.

**Comment by pacak** [comment]
I'll merge this and will work a bit on error messages, I can release a new version as soon as you confirm it works for you.

Here's how I'm using it in some of my other crates: https://github.com/pacak/hackerman/pull/8/files
generated documentation contains more detailed explanations from doc comments.

**Comment by ematipico** [comment]
Thank you, lot of goodies coming 🚀

**Pull request merged**